hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java

```diff
@@ -2130,7 +2135,30 @@ public ExchangeRates midnightRates() {
 		return state.tokenAssociations();
 	}
 
-	public FCInvertibleHashMap<MerkleUniqueTokenId, MerkleUniqueToken, OwnerIdentifier> uniqueTokens() {
+	public void rebuildOwnershipsAndAssociations() {
+		var uniqueTokens= state.uniqueTokens();
+		var keys = uniqueTokens.keySet();
+		for (MerkleUniqueTokenId key : keys) {
+			this.uniqueTokenAssociations.associate(key.tokenId(), key);
+			this.uniqueTokenOwnership.associate(uniqueTokens.get(key).getOwner(), key);
+		}
+	}
+
+	public FCOneToManyRelation<EntityId, MerkleUniqueTokenId> uniqueTokenAssociations() {
+		if (this.uniqueTokenAssociations == null) {
+			rebuildOwnershipsAndAssociations();
+		}
+		return this.uniqueTokenAssociations;
+	}
+
+	public FCOneToManyRelation<EntityId, MerkleUniqueTokenId> uniqueTokenAccountOwnerships() {
```

Comment:
We will have to have them as `AtomicReference<FCOneToManyRelation<EntityId, MerkleUniqueTokenId>> queriableTokenAccountOwnerships`